app/components/offline_indicator/offline_indicator.js

```diff
@@ -53,20 +46,37 @@ export default class OfflineIndicator extends PureComponent {
     }
 
     componentWillReceiveProps(nextProps) {
-        const {appState, isConnecting, isOnline, webSocketStatus} = nextProps;
-        if (appState) { // The app is in the foreground
-            if (isOnline) {
-                if (this.state.network && webSocketStatus === RequestStatus.SUCCESS) {
-                    // Show the connected animation only if we had a previous network status
-                    this.connected();
-                } else if ((webSocketStatus === RequestStatus.STARTED || webSocketStatus === RequestStatus.FAILURE) && isConnecting) {
-                    // Show the connecting bar if it failed to connect at least twice
-                    this.connecting();
-                }
-            } else {
-                this.offline();
+        const {isConnecting, isOnline, webSocketStatus} = nextProps;
+        if (isOnline) {
+            if (this.state.network && this.props.webSocketStatus === RequestStatus.STARTED && webSocketStatus === RequestStatus.SUCCESS) {
+                // Show the connected animation only if we had a previous network status
+                this.connected();
+            } else if (this.props.webSocketStatus === RequestStatus.STARTED && webSocketStatus === RequestStatus.FAILURE && isConnecting) {
+                // Show the connecting bar if it failed to connect at least twice
+                this.connecting();
             }
+        } else {
+            this.offline();
+        }
+    }
+
+    shouldComponentUpdate(nextProps) {
```

Comment:
I'm not sure I understand this. Couldn't we just check if `this.state.network` changes instead of these 3 cases?

Reply:
oh yes, good catch... makes total sense